app/panel/components/BuildingBlocks/GhosteryFeature.jsx

```diff
@@ -14,83 +14,125 @@
-		this.getRestrictText = this.getRestrictText.bind(this);
+		this.handleClick = this.handleClick.bind(this);
 	}
 
 	/**
-	* Gets the text for the Trust Button under different conditions
-	* @return {String} The text for the Trust Button as a string
-	*/
-	getTrustText() {
-		if (this.props.isCondensed) {
-			return '';
-		} else if (this.props.sitePolicy === 2) {
-			return t('summary_trust_site_active');
+	 * Handles user click on the Ghostery Feature button
+	 */
+	handleClick() {
+		if (this.props.blockingPausedOrDisabled) {
+			return;
 		}
-		return t('summary_trust_site');
+
+		this.props.handleClick(this.props.type);
 	}
 
-	/**
-	* Gets the text for the Restrict Button under different conditions
-	* @return {String} The text for the Restrict Button as a string
-	*/
-	getRestrictText() {
-		if (this.props.isCondensed) {
+	_getButtonText(sitePolicy, showText, type) {
+		if (!showText) {
 			return '';
-		} else if (this.props.sitePolicy === 1) {
-			return t('summary_restrict_site_active');
 		}
-		return t('summary_restrict_site');
-	}
 
-	/**
-	 * Handles the click event for the Trust Site button
-	 */
-	clickTrustButton() {
-		if (this.props.isInactive) {
-			return;
+		switch (type) {
+			case 'trust':
+				return (sitePolicy === WHITELISTED ?
+					t('summary_trust_site_active') :
+					t('summary_trust_site'));
+			case 'restrict':
+				return (sitePolicy === BLACKLISTED ?
+					t('summary_restrict_site_active') :
+					t('summary_restrict_site'));
+			default:
+				return 'Check button type you are passing to GhosteryFeature for typos and make sure it is being handled by getButtonText';
 		}
-		this.props.clickButton('trust');
 	}
 
-	/**
-	 * Handles the click event for the Custom Settings button
-	 */
-	clickCustomButton() {
-		if (this.props.isInactive) {
-			return;
+	_getTooltipText(sitePolicy, type) {
+		switch (type) {
+			case 'trust':
+				return (sitePolicy === WHITELISTED ?
+					t('tooltip_trust_on') :
+					t('tooltip_trust'));
+			case 'restrict':
+				return (sitePolicy === BLACKLISTED ?
+					t('tooltip_restrict_on') :
+					t('tooltip_restrict'));
+			default:
+				return 'Check button type you are passing to GhosteryFeature for typos and make sure it is being handled by getTooltipText';
 		}
-		this.props.clickButton('custom');
 	}
 
-	/**
-	 * Handles the click event for the Restrict Site button
-	 */
-	clickRestrictButton() {
-		if (this.props.isInactive) {
-			return;
+	_isFeatureActive(type, sitePolicy) {
+		switch (type) {
+			case 'trust':
+				return sitePolicy === WHITELISTED;
+			case 'restrict':
+				return sitePolicy === BLACKLISTED;
+			default:
+				return false;
 		}
-		this.props.clickButton('restrict');
+	}
+
+	render() {
+		const {
+			blockingPausedOrDisabled,
+			narrow,
+			short,
+			showText,
+			sitePolicy,
+			tooltipPosition,
+			type
+		} = this.props;
+
+		const active = this._isFeatureActive(type, sitePolicy);
+		// TODO Foundation dependency: button
```

Comment:
Why is a foundation dependency listed as a ToDo?

Reply:
Leaving as is to help locate Foundation dependencies in case we move forward with removing Foundation in the future.